components/src/dynamo/vllm/state_agent.py

```
@@ -0,0 +1,198 @@
+
+    endpoint = ZmqEventPublisher.offset_endpoint_port(base, data_parallel_rank=rank)
+    if not endpoint.startswith("tcp://") or ":" not in endpoint[6:]:
+        raise ValueError(f"unsupported vLLM KV event endpoint: {endpoint!r}")
+    port = endpoint.rsplit(":", 1)[1]
+    if not port.isdigit():
+        raise ValueError(f"vLLM KV event endpoint has no numeric port: {endpoint!r}")
+    return f"tcp://{host}:{port}"
+
+
+def _validate_advertise_host(host: str) -> None:
+    if host in {"*", "localhost", "0.0.0.0", "::", "[::]"}:
```

Ruff (0.16.1)

[error] Possible binding to all interfaces

(S104)